### PR TITLE
Un-deprecate default floating point Orderings

### LINKED ISSUE #11844: List(1.0, -1.0).sorted gives a deprecation warning
_Originally posted by @odersky in https://github.com/scala/bug/issues/10511#issuecomment-569510889_

Ref scala/scala#6323
Ref scala/scala#6410

## steps

```scala
scala> List(1.0, -1.0).sorted
warning: there was one deprecation warning (since 2.13.0); for details, enable `:setting -deprecation' or `:replay -deprecation'
res0: List[Double] = List(-1.0, 1.0)
```

## problem

It gives a deprecation warning. I think this is _very bad_, not to say catastrophic.

## expectation

These things should just work. I don't care what ordering is used and whether it is inconsistent or not (we all know IEEE is broken and unfixable). But we should do the right thing for everyday users. As an everyday user I do not care what the ordering does for NaN and -0.0, maybe I do not even know that these weird numbers exist! (and they should not exist, if I go by common mathematical logic). But I _do_ care that I can sort a list of doubles without going into the deeper mysteries of Scala's implicit system. This requirement is now violated. 

## notes

I think we should revert ASAP. We should keep the old total ordering and offer IEEE as an optional alternative that needs to be imported explicitly.

### PR DESCRIPTION
**Pull request by NthPortal**
Un-deprecate default `Ordering`s for `Float` and `Double`,
and change their behaviors to be a combination of those
of `TotalOrdering` and `IeeeOrdering`.

Implements Sébastien's suggestion in https://github.com/scala/bug/issues/11844#issuecomment-570322102 (as interpreted by me).

Fixes scala/bug#11844

### CODE REVIEW

**Review by Ichoran — COMMENTED** [review]
I don't think this is the best solution, but it looks correctly implemented (and clearly documented) if this is the solution we decide on.

File: src/library/scala/math/Ordering.scala [hunk 1]
```diff
@@ -482,10 +501,29 @@ object Ordering extends LowPriorityOrderingImplicits {
...
+    * (as in both `TotalOrdering` and `IeeeOrdering`).
+    *   - `lt`, `lteq`, `equiv`, `gteq` and `gt` are consistent with `compare`,
+    * and treat `NaN` as larger than all other `Double` values (as in `TotalOrdering`).
+    *
+    * It is preferable, rather than using this `Ordering`, to use `TotalOrdering`
+    * or `IeeeOrdering` (by using a local import, assigning an implicit val, or
+    * passing one explicitly) depending on the desired behavior.
+    */
+  // No longer deprecated; see scala/bug#11844
+  implicit object DeprecatedDoubleOrdering extends Ordering[Double] {
```
Comment (NthPortal) [inline src/library/scala/math/Ordering.scala:522]: it needs to still extend `Double.TotalOrdering`, for bincompat reasons. yay =/

**Comment by SethTisue** [comment]
We need to pick a solution for 2.13.2. ~~This seems like the leading contender (agree, @eed3si9n?).~~ (gah, I confused it with #8612)

But as per the discussion on https://github.com/scala/bug/issues/11844 , the deprecation warning should become a migration warning; @NthPortal are you interested in doing that in the next week or so? If you're too busy, it's fine, one of us from the core team can do it.

**Comment by eed3si9n** [comment]
> We need to pick a solution for 2.13.2. This seems like the leading contender (agree, @eed3si9n?).

Not exactly. This PR is neither Total nor Ieee, but it implements a hybrid proposed by @sjrd in https://github.com/scala/bug/issues/11844#issuecomment-570322102

> 1. Un-deprecate `DeprecatedOrdering`
> 2. Change the implementation of `x min y` and `x max y` so that the special-case `{Float,Double}.DeprecatedOrdering`, inn order to apply IEEE behavior.

I don't think we've really discussed the tradeoff of special casing `min` and `max` to be inconsistent with `compare`. If we are doing `TotalOrdering`, I think we should do that for all methods.

**Comment by SethTisue** [comment]
I think we can close this PR, as I believe it is too late to change the runtime behavior in the 2.13.x series.

**Pull request closed without merging**